tests/accessor_legacy/accessor_api_image_common.h

```diff
@@ -1834,6 +1643,7 @@ class check_image_accessor_api_type {
     check_image_accessor_api_dim<T, 1>(log, typeName, count, size, queue);
     check_image_accessor_api_dim<T, 2>(log, typeName, count, size, queue);
     check_image_accessor_api_dim<T, 3>(log, typeName, count, size, queue);
```

Comment:
`check_image_accessor_api_dim` template is removed, so this code doesn't compile.
@andylshort, please, clean-up all dependent code. 
Here if we remove `check_image_accessor_api_dim` here, `check_image_accessor_api_type` doesn't do anything useful anymore, so we need update it and its users.

Reply:
This is part of why I commented them out as I was unsure how to address this dependent code. The dependencies on this are `accessor_api_image_core.cpp` and `accessor_api_image_fp16`. Without the non-compliant image accessor code that's causing other tests to fail, as you said these also don't do anything useful anymore. Valid argument for removal of these legacy tests?

Reply:
Yes. I think we should remove `check_image_accessor_api_type` and all invocations of this template. If callers of this template become no-op, we should remove them too.